csrc/scheduler/matmul.cpp

```diff
@@ -997,34 +1020,43 @@ void scheduleMatmul(Fusion* fusion, const MatmulParams& params) {
   acr->axis(-1)->parallelize(ParallelType::Vectorize);
   bcr->axis(-1)->parallelize(ParallelType::Vectorize);
 
-  //  0  1  2  3  4   5   6   7  8  9  10 11
-  // [B Mo No Ko Kwo Mwo Nwo Mw Nw (Mi Ni Ki)]
+  // batch +      1   2   3   4    5    6    7    8     9    10    11  12
+  //      -13 -12 -11 -10  -9   -8   -7   -6   -5    -4    -3    -2  -1
+  // [..., Kf, Mo, No, Ko, Kw, Mwo, Nwo, Mwi, Nwi, MNi1, MNi2, MNi3, Ki]
+  NVF_ERROR(
+      params.splitk_factor == 1 || num_batch_dims == 0,
+      "Splitk not supported with batch matmul");
   if (num_batch_dims != 0) {
-    mma_result->axis(0)->parallelize(ParallelType::BIDz);
+    mma_result->axis(-14)->parallelize(ParallelType::BIDz);
+  } else if (params.splitk_factor > 1) {
+    mma_result->axis(-13)->parallelize(ParallelType::BIDz);
   }
```

Comment:
What if I want to split-k on a batched matmul? Could we just:
```C++
for (auto _ : c10::irange(num_batch_dims + (params.splitk_factor > 1) - 1) {
  mma_result->merge(0);
}
mma_result->axis(0)->parallelize(ParallelType::BIDz);
```
so that both our split-k supports both bmm and mm?

Reply:
I'll do that. I was hesitant to merge the reduction domain since we will be customizing that reduction in follow-up PRs, but we need to support it so might as well go ahead with it.

Reply:
Actually no, we can't do this can we? If we have batch dimensions they will be `Iteration`, but the splitK dimension is `Reduction`.

Reply:
I could potentially merge them with the M and N outer blocks which are currently mapped to `BIDx` and  `BIDy` ... I will play around and report back.

Reply:
I see. Then probably it is not a thing for this PR...